hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/common/TestHoodieSparkEngineContext.java

```diff
@@ -48,4 +48,16 @@ void testAddRemoveCachedDataIds() {
     assertEquals(Arrays.asList(1, 2, 3), context.removeCachedDataIds(HoodieDataCacheKey.of(basePath, instantTime)));
     assertTrue(context.getCachedDataIds(HoodieDataCacheKey.of(basePath, instantTime)).isEmpty());
   }
+
+  @Test
+  void testSetJobStatus() {
+    // Test data
+    String activeModule = "TestModule";
+    String activityDescription = "Running test operation";
+    String expectedJobDescription = String.format("%s:%s", activeModule, activityDescription);
```

Comment:
Let's add more validation before and after calling `context.setJobStatus`, on both `spark.jobGroup.id` and `spark.job.description` properties.